pkg/index/indexscanner/scanner.go

```diff
@@ -31,37 +31,35 @@ import (
 )
 
 // LoadPluginListFromFS will parse and retrieve all plugin files.
-func LoadPluginListFromFS(indexDir string) (index.PluginList, error) {
-	var indexList index.PluginList
+func LoadPluginListFromFS(indexDir string) ([]index.Plugin, error) {
 	indexDir, err := filepath.EvalSymlinks(indexDir)
 	if err != nil {
-		return indexList, err
+		return nil, err
 	}
 
 	files, err := ioutil.ReadDir(filepath.Join(indexDir, "plugins"))
 	if err != nil {
-		return indexList, errors.Wrap(err, "failed to open index dir")
+		return nil, errors.Wrap(err, "failed to open index dir")
 	}
 
+	list := make([]index.Plugin, 0)
```

Comment:
I find it good practice to pre-allocate the required array like that:
```suggestion
	list := make([]index.Plugin, 0, len(files))
```
Granted, it's a bit too large but a good heuristic.

Reply:
I honestly don't like the static analysis tool that made me do this.
I want to just do `var list []type` what's wrong with that if I don't know the exact size? Any ideas why it's complaining?

Reply:
Not 100% sure here. But I think it's because the linter assumes the `list` ought to be approximately the size of `files`. And in that case it makes perfect sense to pre-allocate.